libs/ui/elements/src/lib/pagination-buttons/pagination-buttons.component.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  Output,
+} from '@angular/core'
+
+@Component({
+  selector: 'gn-ui-pagination-buttons',
+  templateUrl: './pagination-buttons.component.html',
+  styleUrls: ['./pagination-buttons.component.css'],
+})
+export class PaginationButtonsComponent implements OnChanges {
+  @Input() currentPage: number
+  @Input() totalPages: number
+  visiblePages: (number | '...')[] = []
+  @Output() newCurrentPageEvent = new EventEmitter<number>()
+
+  ngOnChanges(): void {
+    this.calculateVisiblePages()
+  }
+
+  calculateVisiblePages(): void {
+    const maxVisiblePages = 5
+    const halfVisible = Math.floor(maxVisiblePages / 2)
+    const startPage = Math.max(this.currentPage - halfVisible, 1)
+    const endPage = Math.min(startPage + maxVisiblePages - 1, this.totalPages)
```

Comment:
```suggestion
    const endPage = Math.min(this.currentPage + halfVisible, this.totalPages)
```

Otherwise when on page 1, the pages 1 to 5 are shown :)

Reply:
Thanks, that's a good idea!